beacon-chain/types/attestation.go

```diff
@@ -121,3 +122,43 @@ func (a *Attestation) ObliqueParentHashes() [][32]byte {
 func (a *Attestation) AggregateSig() []uint64 {
 	return a.data.AggregateSig
 }
+
+// VerifyAttestation verifies the proposer's attestation of the block.
+// Proposers broadcast the attestation along with the block to its peers.
+func (a *Attestation) VerifyAttestation(proposerShardID uint64) error {
```

Comment:
I expect this function to take `proposerIndex` as well to ensure the correct validator created/signed the message. I see the TODO for future sig check, but I would pass the param in now to get the function signature correct.